src/components/types/GameType.tsx

```diff
@@ -0,0 +1,10 @@
+export type CellType = number | null;
+export type CellsType = CellType[][];
+export type HistoryType = CellsType[];
+export type DirectionType = 'up' | 'left' | 'right' | 'down';
+export type RotateDegreeType = 0 | 90 | 180 | 270;
+export enum GameOverStatus {
+  Success = 'success',
+  Fail = 'fail',
+  None = 'none',
+}
```

Comment:
```suggestion
export enum GameOverStatus {
  SUCCESS = 'SUCCESS',
  FAIL = 'FAIL',
  NONE = 'NONE',
}
```

enum 네이밍 컨벤션은 보통 키값에 UPPER_CASE, value는 키와 동일하게 작성합니다

value가 key와 동일해야 javascript로 트랜스파일했을 때 좀더 깔끔해서 그런건데, 엄청 strict한 규칙은 아니고 그냥 일반적으로 그렇다~ 정도입니다!